src/app.py

```diff
@@ -1,109 +1,126 @@
+                style={'text_aligh': 'left', 'color': '#0f3c63', 'font-family': 'sans-serif'}),
+           html.Br(),
+           dcc.RangeSlider(
+               className="slider_class",
+                id="year_range",
+                count=1,
+                step=1,
+                min=1980,
+                max=2016,
+                value=[2000, 2005],
+                marks={1980: {"label": "1980"}, 2016: {"label": "2016"}}
+                ),
+            html.Br(),
+            html.Div(html.P("Select the movie genre"),
+            style={'text_aligh': 'left', 'color': '#0f3c63', 'font-family': 'sans-serif'}),
+            html.Br(),
+            dcc.Checklist(
+                id="genre_checklist",                    
+                className="genre-container",
+                inputClassName="genre-input",
+                labelClassName="genre-label",
+                options=[{"label": i, "value": i} for i in genre],                        
+                value=default_genres,
+                labelStyle={"display":"block",
+                            "margin-left": "30px"}),
+            ])
+            ])
+            ])
+        ]),
+    html.Br(), 
+    #plot graphs
+   dbc.Col(
+        html.Iframe(
+            id="plot_graphs",
+            style={"display": "block",
+                    "margin": "auto",
+                    "overflow": " hidden", 
+                    "width" : "300%", "height":"800px"}
+        ))
+        ])
+])
+
+@app.callback(
+    Output('plot_graphs', 'srcDoc'),
+    Input('genre_checklist', 'value'),
+    Input('year_range', 'value')
+)
+
+#plotting function
+def plot_graphs(genre, year_range):
+    filter_genre = df[df.Major_genre.isin(genre)]
+
+    filter_data = filter_genre.query("Year >= @year_range[0] & Year <= @year_range[1] & Major_genre in @genre").dropna()
+    brush = alt.selection_interval()
+    click = alt.selection_multi(fields=['Major_genre'])
+
+    scatter = alt.Chart(filter_data, title="IMDB Rating Vs. Duration (in mins)").mark_circle(opacity=0.7).encode(
+        x=alt.Y("IMDB_rating", title="IMDB Rating"),
+        y=alt.X("Duration", title="Duration (in mins)"),
+        color=alt.condition(brush, 'Major_genre', alt.value('lightgray')),
+        tooltip=alt.Tooltip(["IMDB_rating", "Duration"]),
+        opacity=alt.condition(click, alt.value(0.9), alt.value(0.1))).add_selection(brush).properties(
+        width=250,
+        height=200
+    )
+
+    bar = alt.Chart(filter_data, title='Gross revenue (box office) by genre').mark_bar().encode(
+        x=alt.X('sum(US_Revenue)', axis=alt.Axis(title='Gross Revenue (in millions USD)')),
+        y=alt.Y('Major_genre', axis=alt.Axis(title='Major genre')),
+        color=alt.Color('Major_genre'))
+    barplot = bar.encode(opacity=alt.condition(click, alt.value(0.9), alt.value(0.1)),
+       tooltip=alt.Tooltip('sum(US_Revenue)', format="$,.0f")).add_selection(click).properties(
+        width=550,
+        height=180

```

Comment:
```suggestion
        height=200

```
to have even hight for upper and lower parts